### PR TITLE
Add IPFS signature to polls and votes

### PR DESCRIPTION
**Pull request by ashutoshpw**
Zenhub Ticket [#781](https://app.zenhub.com/workspaces/tezos-homebase-63238426ddf385159a4c4b12/issues/gh/dorgtech/homebase-app/781)

### CODE REVIEW

File: utils.js [hunk 1]
```diff
@@ -137,63 +129,48 @@ const getUserTotalVotingPowerAtReferenceBlock = async (
   daoContract,
   tokenID,
   level,
-  userAddress,
-  isXTZ = false
```
Comment (Man-Jain) [inline utils.js:141]: This should not be removed, i think some older code written previously is overwriting some latest changes.

File: components/choices/index.js [hunk 2]
```diff
@@ -105,7 +111,40 @@ const updateChoiceById = async (req, response) => {
     if (total.eq(0)) {
       throw new Error("No balance at proposal level");
     }
+    const isVoted = await db_connect
```
Comment (Man-Jain) [inline components/choices/index.js:114]: Are these changes relevant to this PR ?

**Review by Man-Jain — CHANGES_REQUESTED** [review]

File: components/choices/index.js (hunk 2 above)
Comment (ashutoshpw) [reply components/choices/index.js:114]: @Man-Jain I copied over some of the changes from the [original PR c](https://github.com/dOrgTech/homebase-lite-backend/pull/17/files#diff-51955b00ac2104e971f4966afbcbf92b6dab10645c827d0ceed8838410d90649)reated for this. Please do let me know what should I remove from here?

File: services/ipfs.service.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const AppConfig = require('../config')
+
+const { NFTStorage } = require('nft.storage')
+const NFT_STORAGE_KEY = AppConfig.IPFS.NFT_STORAGE_TOKEN
+
+async function uploadToIPFS(jsonData) {
+    const data = new Blob([jsonData])
+    const client = new NFTStorage({ token: NFT_STORAGE_KEY })
+    const cid = await client.storeBlob(data)
```
[the remark this answers is not in the document]
Comment (ashutoshpw) [reply services/ipfs.service.js:9]: Yes, I will modify the code to store it as JSON.

File: utils.js (hunk 1 above)
Comment (ashutoshpw) [reply utils.js:141]: On it. Will fix this soon.

File: components/polls/index.js [hunk 4]
```diff
@@ -141,11 +144,29 @@ const addPoll = async (req, response) => {
...
 
+    const pollExists = await db_connect
+      .collection("Polls")
+      .findOne({ payloadBytes });
+
+    if (pollExists) {
+      throw new Error("Invalid Signature, Poll already exists");
+    }
+
+    const cidLink = await uploadToIPFS(
```
[the remark this answers is not in the document]
Comment (ashutoshpw) [reply components/polls/index.js:159]: In this PR it seems like we are already throwing an error if the payload is used again. So spoofing might  not be possible, can you please look at this again?

File: services/ipfs.service.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashutoshpw) [reply services/ipfs.service.js:9]: It is now being uploaded as plain text.

**Review by Man-Jain — APPROVED** [review]
